### ISSUE TITLE
Adw.EntryRow not available

### ISSUE THREAD

**Issue by joshuawood2894** [diamondburned/gotk4-adwaita#3]
The generated bindings don't seem to include Adwaita's EntryRow component. Is this something that could be added?

**Comment by diamondburned** [comment]
Perhaps trying to `go generate` with `GIR_VERBOSE=1` would help debug the issue.

**Comment by zeitue** [comment]
It seems that the EntryRow doesn't show up when generating the bindings.

```bash
[nix-shell:/gotk4-adwaita]# GIR_VERBOSE=1 go generate         
2022/09/19 20:51:32 added from package gobject-introspection-1.0 namespace GLib-2 /nix/store/6a0z6xdhm3acm74sn9bh0a56hmqibwp6-gobject-introspection-1.70.0-dev/share/gir-1.0/GLib-2.0.gir
2022/09/19 20:51:32 added from package gobject-introspection-1.0 namespace GObject-2 /nix/store/6a0z6xdhm3acm74sn9bh0a56hmqibwp6-gobject-introspection-1.70.0-dev/share/gir-1.0/GObject-2.0.gir
2022/09/19 20:51:32 added from package gobject-introspection-1.0 namespace Gio-2 /nix/store/6a0z6xdhm3acm74sn9bh0a56hmqibwp6-gobject-introspection-1.70.0-dev/share/gir-1.0/Gio-2.0.gir
2022/09/19 20:51:32 added from package gobject-introspection-1.0 namespace cairo-1 /nix/store/6a0z6xdhm3acm74sn9bh0a56hmqibwp6-gobject-introspection-1.70.0-dev/share/gir-1.0/cairo-1.0.gir
2022/09/19 20:51:32 added from package gdk-pixbuf-2.0 namespace GdkPixbuf-2 /nix/store/mpjbjnsqp40k5x6my8m9lvnjfxvs5vci-gdk-pixbuf-2.42.6-dev/share/gir-1.0/GdkPixbuf-2.0.gir
2022/09/19 20:51:32 added from package gdk-pixbuf-2.0 namespace GdkPixdata-2 /nix/store/mpjbjnsqp40k5x6my8m9lvnjfxvs5vci-gdk-pixbuf-2.42.6-dev/share/gir-1.0/GdkPixdata-2.0.gir
2022/09/19 20:51:32 added from package graphene-1.0 namespace Graphene-1 /nix/store/71j5aarpf2h39g21cjxld6s9x7my3qfm-graphene-1.10.6/share/gir-1.0/Graphene-1.0.gir
2022/09/19 20:51:32 added from package atk namespace Atk-1 /nix/store/sjif6yf5pi0xwbf8zmqv2jsrwcfc8v84-atk-2.36.0-dev/share/gir-1.0/Atk-1.0.gir
2022/09/19 20:51:32 added from package pango namespace Pango-1 /nix/store/fckl2g2rl969mar66g5mvk051149jmzv-pango-1.50.3-dev/share/gir-1.0/Pango-1.0.gir
2022/09/19 20:51:32 added from package pango namespace PangoCairo-1 /nix/store/fckl2g2rl969mar66g5mvk051149jmzv-pango-1.50.3-dev/share/gir-1.0/PangoCairo-1.0.gir
2022/09/19 20:51:32 added from package gtk4 namespace Gdk-4 /nix/store/yy2wskx76152q0riyhmv780s2x6bbc7a-gtk4-4.6.0-dev/share/gir-1.0/Gdk-4.0.gir
2022/09/19 20:51:32 added from package gtk4 namespace GdkWayland-4 /nix/store/yy2wskx76152q0riyhmv780s2x6bbc7a-gtk4-4.6.0-dev/share/gir-1.0/GdkWayland-4.0.gir
2022/09/19 20:51:32 added from package gtk4 namespace GdkX11-4 /nix/store/yy2wskx76152q0riyhmv780s2x6bbc7a-gtk4-4.6.0-dev/share/gir-1.0/GdkX11-4.0.gir
2022/09/19 20:51:32 added from package gtk4 namespace Gsk-4 /nix/store/yy2wskx76152q0riyhmv780s2x6bbc7a-gtk4-4.6.0-dev/share/gir-1.0/Gsk-4.0.gir
2022/09/19 20:51:32 added from package gtk4 namespace Gtk-4 /nix/store/yy2wskx76152q0riyhmv780s2x6bbc7a-gtk4-4.6.0-dev/share/gir-1.0/Gtk-4.0.gir
2022/09/19 20:51:32 added from package gtk+-3.0 namespace Gdk-3 /nix/store/zmsvipsngcbwg9n6abz0zr7nrj9sbh59-gtk+3-3.24.31-dev/share/gir-1.0/Gdk-3.0.gir
2022/09/19 20:51:32 added from package gtk+-3.0 namespace GdkX11-3 /nix/store/zmsvipsngcbwg9n6abz0zr7nrj9sbh59-gtk+3-3.24.31-dev/share/gir-1.0/GdkX11-3.0.gir
2022/09/19 20:51:32 added from package gtk+-3.0 namespace Gtk-3 /nix/store/zmsvipsngcbwg9n6abz0zr7nrj9sbh59-gtk+3-3.24.31-dev/share/gir-1.0/Gtk-3.0.gir
2022/09/19 20:51:32 added from package libadwaita-1 namespace Adw-1 /nix/store/1z3hwr8djpiz7x0lxhg3hl51min8rsqq-libadwaita-1.0.1-dev/share/gir-1.0/Adw-1.gir
girgen: skipped: package adw/v1: function easing_ease
girgen: debug: package adw/v1: callable  (C.adw_toast_set_action_target) setMethods skipped adw_toast_set_action_target
girgen: debug: package adw/v1: record ActionRowClass (C.AdwActionRowClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record ActionRowClass
girgen: debug: package adw/v1: record AnimationClass (C.AdwAnimationClass) disguised
girgen: skipped: package adw/v1: record AnimationClass
girgen: debug: package adw/v1: record AnimationTargetClass (C.AdwAnimationTargetClass) disguised
girgen: skipped: package adw/v1: record AnimationTargetClass
girgen: debug: package adw/v1: record ApplicationClass (C.AdwApplicationClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record ApplicationClass
girgen: debug: package adw/v1: record ApplicationWindowClass (C.AdwApplicationWindowClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record ApplicationWindowClass
girgen: debug: package adw/v1: record AvatarClass (C.AdwAvatarClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record AvatarClass
girgen: debug: package adw/v1: record BinClass (C.AdwBinClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record BinClass
girgen: debug: package adw/v1: record ButtonContentClass (C.AdwButtonContentClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record ButtonContentClass
girgen: debug: package adw/v1: record CallbackAnimationTargetClass (C.AdwCallbackAnimationTargetClass) disguised
girgen: skipped: package adw/v1: record CallbackAnimationTargetClass
girgen: debug: package adw/v1: record CarouselClass (C.AdwCarouselClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record CarouselClass
girgen: debug: package adw/v1: record CarouselIndicatorDotsClass (C.AdwCarouselIndicatorDotsClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record CarouselIndicatorDotsClass
girgen: debug: package adw/v1: record CarouselIndicatorLinesClass (C.AdwCarouselIndicatorLinesClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record CarouselIndicatorLinesClass
girgen: debug: package adw/v1: record ClampClass (C.AdwClampClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record ClampClass
girgen: debug: package adw/v1: record ClampLayoutClass (C.AdwClampLayoutClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record ClampLayoutClass
girgen: debug: package adw/v1: record ClampScrollableClass (C.AdwClampScrollableClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record ClampScrollableClass
girgen: debug: package adw/v1: record ComboRowClass (C.AdwComboRowClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record ComboRowClass
girgen: debug: package adw/v1: record EnumListItemClass (C.AdwEnumListItemClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record EnumListItemClass
girgen: debug: package adw/v1: record EnumListModelClass (C.AdwEnumListModelClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record EnumListModelClass
girgen: debug: package adw/v1: record ExpanderRowClass (C.AdwExpanderRowClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record ExpanderRowClass
girgen: debug: package adw/v1: record FlapClass (C.AdwFlapClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record FlapClass
girgen: debug: package adw/v1: record HeaderBarClass (C.AdwHeaderBarClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record HeaderBarClass
girgen: debug: package adw/v1: record LeafletClass (C.AdwLeafletClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record LeafletClass
girgen: debug: package adw/v1: record LeafletPageClass (C.AdwLeafletPageClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record LeafletPageClass
girgen: debug: package adw/v1: record PreferencesGroupClass (C.AdwPreferencesGroupClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record PreferencesGroupClass
girgen: debug: package adw/v1: record PreferencesPageClass (C.AdwPreferencesPageClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record PreferencesPageClass
girgen: debug: package adw/v1: record PreferencesRowClass (C.AdwPreferencesRowClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record PreferencesRowClass
girgen: debug: package adw/v1: record PreferencesWindowClass (C.AdwPreferencesWindowClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record PreferencesWindowClass
girgen: debug: package adw/v1: record SplitButtonClass (C.AdwSplitButtonClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record SplitButtonClass
girgen: debug: package adw/v1: record SpringAnimationClass (C.AdwSpringAnimationClass) disguised
girgen: skipped: package adw/v1: record SpringAnimationClass
girgen: debug: package adw/v1: callable  (C.adw_spring_params_ref) not generating ignored name ref
girgen: skipped: package adw/v1: record SpringParams (C.AdwSpringParams): record SpringParams method ref
girgen: debug: package adw/v1: callable  (C.adw_spring_params_unref) not generating ignored name unref
girgen: skipped: package adw/v1: record SpringParams (C.AdwSpringParams): record SpringParams method unref
girgen: debug: package adw/v1: record SqueezerClass (C.AdwSqueezerClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record SqueezerClass
girgen: debug: package adw/v1: record SqueezerPageClass (C.AdwSqueezerPageClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record SqueezerPageClass
girgen: debug: package adw/v1: record StatusPageClass (C.AdwStatusPageClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record StatusPageClass
girgen: debug: package adw/v1: record StyleManagerClass (C.AdwStyleManagerClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record StyleManagerClass
girgen: debug: package adw/v1: record SwipeTrackerClass (C.AdwSwipeTrackerClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record SwipeTrackerClass
girgen: debug: package adw/v1: record SwipeableInterface (C.AdwSwipeableInterface) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record SwipeableInterface
girgen: debug: package adw/v1: record TabBarClass (C.AdwTabBarClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record TabBarClass
girgen: debug: package adw/v1: record TabPageClass (C.AdwTabPageClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record TabPageClass
girgen: debug: package adw/v1: record TabViewClass (C.AdwTabViewClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record TabViewClass
girgen: debug: package adw/v1: record TimedAnimationClass (C.AdwTimedAnimationClass) disguised
girgen: skipped: package adw/v1: record TimedAnimationClass
girgen: debug: package adw/v1: record ToastClass (C.AdwToastClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record ToastClass
girgen: debug: package adw/v1: record ToastOverlayClass (C.AdwToastOverlayClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record ToastOverlayClass
girgen: debug: package adw/v1: record ViewStackClass (C.AdwViewStackClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record ViewStackClass
girgen: debug: package adw/v1: record ViewStackPageClass (C.AdwViewStackPageClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record ViewStackPageClass
girgen: debug: package adw/v1: record ViewSwitcherBarClass (C.AdwViewSwitcherBarClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record ViewSwitcherBarClass
girgen: debug: package adw/v1: record ViewSwitcherClass (C.AdwViewSwitcherClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record ViewSwitcherClass
girgen: debug: package adw/v1: record ViewSwitcherTitleClass (C.AdwViewSwitcherTitleClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record ViewSwitcherTitleClass
girgen: debug: package adw/v1: record WindowClass (C.AdwWindowClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record WindowClass
girgen: debug: package adw/v1: record WindowTitleClass (C.AdwWindowTitleClass) IsGTypeStructFor or has underscore prefixed
girgen: skipped: package adw/v1: record WindowTitleClass

```

**Comment by diamondburned** [comment]
It's probably because the pinned Nixpkgs doesn't have a new enough libadwaita version. `EntryRow` is added in 1.2, according to the documentation.

gotk4 doesn't have a way to generate backwards-compatible bindings yet, so we have to lock the version at a mildly outdated one, unfortunately.

**Comment by zeitue** [comment]
How would I go about pinning this to a newer libadwaita version? I'm not very familiar with Nixpkgs or NixOS.

**Comment by diamondburned** [comment]
A quick hack is adding the following into `.nix/overlay.nix`:

```nix
self: super: {
	libadwaita = (import <nixpkgs> {}).libadwaita;
}
```

However you may (and should) import a newer Nixpkgs version that has the version
of libadwaita that you want, then use that in the overlay somewhat like so:

```nix
self: super:

let laterNixpkgs = import (super.fetchFromGitHub { ... }) {};

in {
	libadwaita = laterNixpkgs.libadwaita;
}
```

**Comment by DeedleFake** [comment]
It should be noted if you're importing this package instead of generating the bindings yourself, this can be solved by simply using `adw-1.2` as the version in your `go.mod` by running

```bash
$ go get github.com/diamondburned/gotk4-adwaita/pkg@adw-1.2
```

inside of your module.

**Comment by diamondburned** [comment]
Adw.EntryRow is now available in gotk4-adwaita as `adw.EntryRow`, so this issue is fixed.
